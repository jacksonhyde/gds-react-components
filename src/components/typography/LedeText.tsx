import React, { HTMLProps } from 'react';
import classNames from 'classnames';

const LedeText: React.FC<HTMLProps<HTMLParagraphElement>> = ({ className, ...rest }) => (
  <p className={classNames('govuk-lede-text', className)} {...rest} />
);

export default LedeText;
