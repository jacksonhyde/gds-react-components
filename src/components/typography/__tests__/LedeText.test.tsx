import React from 'react';
import { shallow } from 'enzyme';
import LedeText from '../LedeText';

describe('LedeText', () => {
  it('matches snapshot', () => {
    const element = shallow(<LedeText>Text</LedeText>);
    expect(element.text()).toBe('Text');
    expect(element).toMatchSnapshot();
    expect(element.hasClass('govuk-lede-text')).toBeTruthy();
    element.unmount();
  });
});
