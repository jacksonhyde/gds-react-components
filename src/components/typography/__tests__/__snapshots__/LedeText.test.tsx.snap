// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LedeText matches snapshot 1`] = `
<p
  className="govuk-lede-text"
>
  Text
</p>
`;
